feature-libs/asm/components/customer-selection/customer-selection.component.ts

```diff
@@ -29,10 +29,10 @@ export class CustomerSelectionComponent implements OnInit, OnDestroy {
   private subscription = new Subscription();
   searchResultsLoading$: Observable<boolean>;
   searchResults: Observable<CustomerSearchPage>;
-  selectedCustomer: User;
+  selectedCustomer: User | undefined;
 
   @Output()
-  submitEvent = new EventEmitter<{ customerId: string }>();
+  submitEvent = new EventEmitter<{ customerId: string | undefined }>();
```

Comment:
Do we need undefined here?
I assume event is emitted only for the user, so looks artificial to me.

Reply:
Let's discuss this one further.  the customer id can be undefined since the field is optional on the `User` interface.

In the AsmMainUiComponent, the case where the customer Id would be false is handles and some feedback to the user is given.  I'm hesitant to refactor this to catch `undefined` here and add a user message in the customer selection component.

Reply:
We agree we leave the `| undefined`